beep/structure.py

```diff
@@ -230,6 +231,12 @@ def get_interpolated_steps(
             )
             if new_df.size == 0:
                 continue
+
+            # If cycle contains a waveform step, interpolate on test_time.
+            if self.data.loc[self.data["cycle_index"] == cycle_index].\
```

Comment:
Having this buried in the structuring can lead to some unexpected behavior since it will silently switch over to time based interpolation if it decides there is a waveform step. Is there a way to make this decision logic more exposed?

Reply:
Yeah, I toggled between putting it where it is now vs one level above, i.e in the `get_interpolated_cycles` method, so that the determination is made prior to *any* regular cycle being interpolated and leaving it at a step-level. We could start out with the second option and see. I am writing up a follow-up ticket to see if there is any unexpected behavior with this later on.